src/Metadata/DepthAwareMetadataInterface.php

```diff
@@ -0,0 +1,23 @@
+<?php
+declare(strict_types=1);
+
+namespace Zend\Expressive\Hal\Metadata;
+
+interface DepthAwareMetadataInterface
+{
+    /**
+     * Returns configured max depth
+     *
+     * @return int
+     */
+    public function getMaxDepth(): int;
```

Comment:
Is this method even necessary? It's never called internally.

Reply:
yes, never called internally. but as we introduce a mechanism which handles depth / maxDepth we should be able to retrieve the maxDepth from the interface.

Reply:
For what use case, though?

The `hasReachedMaxDepth()` method provides _behavior_, which we consume internally. What is the use case for retrieving the max depth setting? What should implementations that ignore depth return in such cases?

Reply:
good point. i guess that method is really useless :O